Content.Server/Weapons/Ranged/Systems/GunSystem.cs

```diff
@@ -174,10 +177,36 @@ private void OnBallisticPrice(EntityUid uid, BallisticAmmoProviderComponent comp
                     break;
                 // Ammo shoots itself
                 case AmmoComponent newAmmo:
-                    shotProjectiles.Add(ent!.Value);
+                    if (ent == null)
+                        break;
+
+                    if (TryComp<ProjectileSpreadComponent>(ent.Value, out var ammoSpreadComp))
+                    {
+                        var spreadEvent = new GunGetAmmoSpreadEvent(ammoSpreadComp.Spread);
+                        RaiseLocalEvent(gunUid, ref spreadEvent);
+
+                        var angles = LinearSpread(mapAngle - spreadEvent.Spread / 2,
+                                     mapAngle + spreadEvent.Spread / 2, ammoSpreadComp.Count);
+
+                        ShootOrThrow(ent.Value, angles[0].ToVec(), gunVelocity, gun, gunUid, user);
+                        shotProjectiles.Add(ent!.Value);
+
+                        for (var i = 1; i < ammoSpreadComp.Count; i++)
+                        {
+                            var newuid = Spawn(ammoSpreadComp.Prototype, fromEnt);
+                            ShootOrThrow(newuid, angles[i].ToVec(), gunVelocity, gun, gunUid, user);
+                            shotProjectiles.Add(ent!.Value);
+                        }
+                    }
+                    else
+                    {
+                        ShootOrThrow(ent.Value, mapDirection, gunVelocity, gun, gunUid, user);
+                        shotProjectiles.Add(ent!.Value);
+                    }
+
```

Comment:
This is slightly too copy-paste with the above for my taste and needs combining please.